### ISSUE TITLE
"Could not find model object to extract residuals." when model is not in global envir

### ISSUE THREAD

**Issue by Deleetdk** [strengejacke/ggeffects#344]
Reprex:

```
library(ggeffects)
library(tidyverse)

lm_model_ok = lm(displ ~ cty, data = mpg)

preds = ggpredict(
  model = lm_model_ok,
  terms = c("cty")
)

#works
plot(preds, residuals = T)

#inside a list
models = list(
  model1 = lm(displ ~ cty, data = mpg)
)

preds = ggpredict(
  model = models$model1,
  terms = c("cty")
)

plot(preds, residuals = T)
#> Warning: Could not find model object to extract residuals.
```

**Comment by Deleetdk** [comment]
Seems to still fail when inside a function.

``` r
library(ggeffects)
library(tidyverse)

lm_model_ok = lm(displ ~ cty, data = mpg)

#global envir
preds = ggpredict(
  model = lm_model_ok,
  terms = "cty"
)

#works
plot(preds, residuals = T)
```

![](https://i.imgur.com/BPmKFJl.png)

``` r

#inside a function
make_residuals_plot = function(model, terms) {
  preds = ggpredict(
    model = model,
    terms = terms
  )

  #works
  plot(preds, residuals = T)
}

make_residuals_plot(lm_model_ok, terms = "cty")
#> Error in if (is.null(model) && grep("\\$|\\[", obj_name)) {: missing value where TRUE/FALSE needed
```

<sup>Created on 2023-07-16 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by strengejacke** [comment]
ok, should be fixed in version `1.2.3.4`.
